src/test/java/com/alipay/oceanbase/rpc/ObTableGroupCommitTest.java

```diff
@@ -0,0 +1,881 @@
+            setGroupCommitSize(group_size);
+
+            // create tasks and threads
+            Runnable[] tasks = new Runnable[op_size];
+            Thread[] threads = new Thread[op_size];
+            for (int i = 0; i < op_size; i++) {
+                tasks[i] = new PutTask(i);
+                if (i == op_size - 1) {
+                    tasks[i] = new FailPutTask(i);
+                }
+                threads[i] = new Thread(tasks[i]);
+            }
+
+            // start thread
+            for (int i = 0; i < op_size; i++) {
+                threads[i].start();
+            }
+
+            // wait thread finish
+            for (int i = 0; i < op_size; i++) {
+                threads[i].join();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            deleteTable(tableName);
+            disableGroupCommit();
+        }
+    }
+
+    @Test
+    // testGroupFail2 many ops are invalid
+    public void testGroupFail2() throws Exception {
+        long group_size = 10;
+        int op_size = 100;
+        try {
+            // set up group commit
+            enableGroupCommit();
+            setGroupCommitSize(group_size);
+
+            // create tasks and threads
+            Runnable[] tasks = new Runnable[op_size];
+            Thread[] threads = new Thread[op_size];
+            for (int i = 0; i < op_size; i++) {
+                tasks[i] = new PutTask(i);
+                if (i % 2 == 0) {
+                    tasks[i] = new FailPutTask(i);
+                }
+                threads[i] = new Thread(tasks[i]);
+            }
+
+            // start thread
+            for (int i = 0; i < op_size; i++) {
+                threads[i].start();
+            }
+
+            // wait thread finish
+            for (int i = 0; i < op_size; i++) {
+                threads[i].join();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            deleteTable(tableName);
+            disableGroupCommit();
+        }
+    }
+
+    @Test
+    // testGroupBanning1 auto ban group commit
+    public void testGroupBanning1() throws Exception {
+        long group_size = 10;
+        int op_size = 1;
+        try {
+            // set up group commit
+            enableGroupCommit();
+            setGroupCommitSize(group_size);
+            setGroupCommitAutoCloseThreshold(1);
+
+            // create tasks and threads
+            Runnable[] tasks = new Runnable[op_size];
+            Thread[] threads = new Thread[op_size];
+            tasks[0] = new FailPutTask(0);
+            threads[0] = new Thread(tasks[0]);
+
+            // start thread
+            for (int i = 0; i < op_size; i++) {
+                threads[i].start();
+            }
+
+            // wait thread finish
+            for (int i = 0; i < op_size; i++) {
+                threads[i].join();
+            }
+
+            assertFalse(isGroupCommitEnable());
```

Comment:
Why not check whether group commit is enabled after `enableGroupCommit()`
We could skip this test once the server does not support the group commit